yt_dlp/extractor/nebula.py

```diff
@@ -0,0 +1,320 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import os
+
+from .common import InfoExtractor
+from ..compat import compat_urllib_parse_unquote, compat_str
+from ..utils import parse_iso8601, ExtractorError, try_get, urljoin, sanitized_Request
+
+
+class NebulaIE(InfoExtractor):
+    """
+    Nebula (https://watchnebula.com/) is a video platform created by the streamer community Standard. It hosts videos
+    off-YouTube from a small hand-picked group of creators.
+
+    All videos require a subscription to watch. There are no known freely available videos. An authentication token to
+    an account with a valid subscription can be specified in multiple ways, including credentials in .netrc or a cookie
+    jar.
+    As neither of these parameters appear to be supported by the unit test runner, it's recommended to set the envvar
+    NEBULA_TOKEN to execute the test runs.
+
+    Nebula uses the Zype video infrastructure and this extractor is using the 'url_transparent' mode to hand off
+    video extraction to the Zype extractor.
+
+    This description has been last updated on 2020-10-22.
+    """
+
+    _VALID_URL = r'https?://(?:www\.)?watchnebula\.com/videos/(?P<id>[-\w]+)'   # the 'id' group is actually the display_id, but we misname it 'id' to be able to use _match_id()

```

Comment:
> the 'id' group is actually the display_id, but we misname it 'id' to be able to use _match_id()

This could potentially cause issues with archive. I'm not sure though and need to investigate further

Reply:
Can you give me a hint (e.g. deep link into docs) for what you refer to by 'archive'?

Would you prefer `_match_id()` not being used for this extractor? It's been a while, this line is a year old, but I think there was some other consequences to not calling that method at all.

Reply:
> Can you give me a hint (e.g. deep link into docs) for what you refer to by 'archive'?

I mean `--download-archive`. I didnt get time to check in-depth whether this needs to be changed. You dont need to change anything for now. This was more of a note to myself that this needs investigation b4 merging